pkg/kubelet/kubelet_node_status.go

```diff
@@ -156,6 +157,32 @@ func (kl *Kubelet) tryRegisterWithApiServer(node *v1.Node) bool {
 	return false
 }
 
+//updateDefaultLabels will set the default labels on the node
+func (kl *Kubelet) updateDefaultLabels(initialNode, existingNode *v1.Node) (needsUpdate bool) {
+	defaultLabels := []string{
+		metav1.LabelHostname,
+		metav1.LabelZoneFailureDomain,
+		metav1.LabelZoneRegion,
+		metav1.LabelInstanceType,
+		metav1.LabelOS,
+		metav1.LabelArch,
+	}
+
+	//Set default labels but make sure to not set labels with empty values
+	for _, label := range defaultLabels {
+		if existingNode.Labels[label] != initialNode.Labels[label] {
+			existingNode.Labels[label] = initialNode.Labels[label]
+			needsUpdate = true
```

Comment:
If it would directly return true, only the first update would be updated

It needs to iterate over all default labels and update them if needed